vm/runtime/src/lib.rs

```diff
@@ -227,3 +251,99 @@ pub enum ConsensusFaultType {
     ParentGrinding = 2,
     TimeOffsetMining = 3,
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use cid::{multihash, Codec};
+    use commcid::FilecoinMultihashCode;
+
+    #[derive(Copy, Debug, Clone)]
+    struct TestSyscalls;
+    impl Syscalls for TestSyscalls {
+        fn verify_signature(
+            &self,
+            _signature: &Signature,
+            _signer: &Address,
+            _plaintext: &[u8],
+        ) -> Result<(), Box<dyn StdError>> {
+            Ok(())
+        }
+        fn hash_blake2b(&self, _data: &[u8]) -> Result<[u8; 32], Box<dyn StdError>> {
+            Ok([0u8; 32])
+        }
+        fn compute_unsealed_sector_cid(
+            &self,
+            _reg: RegisteredProof,
+            _pieces: &[PieceInfo],
+        ) -> Result<Cid, Box<dyn StdError>> {
+            Ok(Default::default())
+        }
+        fn verify_seal(&self, vi: &SealVerifyInfo) -> Result<(), Box<dyn StdError>> {
+            let commd = cid_to_data_commitment_v1(&vi.unsealed_cid)?;
+            let commr = cid_to_replica_commitment_v1(&vi.on_chain.sealed_cid)?;
+            let miner_addr = Address::new_id(vi.sector_id.miner);
+            let miner_payload = miner_addr.payload_bytes();
+            let mut prover_id = ProverId::default();
+            prover_id[..miner_payload.len()].copy_from_slice(&miner_payload);
+
+            if !proofs_verify_seal(
+                vi.on_chain.registered_proof.into(),
+                commr,
+                commd,
+                prover_id,
+                SectorId::from(vi.sector_id.number),
+                vi.randomness,
+                vi.interactive_randomness,
+                &vi.on_chain.proof,
+            )? {
+                return Err(format!(
+                    "Invalid proof detected: {:?}",
+                    base64::encode(&vi.on_chain.proof)
+                )
+                .into());
+            }
+
+            Ok(())
+        }
+        fn verify_post(&self, _vi: &WindowPoStVerifyInfo) -> Result<(), Box<dyn StdError>> {
+            Ok(Default::default())
+        }
+        fn verify_consensus_fault(
+            &self,
+            _h1: &[u8],
+            _h2: &[u8],
+            _extra: &[u8],
+            _earliest: ChainEpoch,
+        ) -> Result<Option<ConsensusFault>, Box<dyn StdError>> {
+            Ok(Some(ConsensusFault {
+                target: Address::new_id(0),
+                epoch: 0,
+                fault_type: ConsensusFaultType::DoubleForkMining,
+            }))
+        }
+    }
```

Comment:
remove all of this, the point is to test the default impl? Can just use the DefaultSyscalls instead of TestSyscalls

Reply:
Moved tests to test dir and switched to use DefaultSyscalls